cvmfs/network/download.cc

```diff
@@ -1413,38 +1335,25 @@ bool DownloadManager::VerifyAndFinalize(const int curl_error, JobInfo *info) {
     LogCvmfs(kLogDownload, kLogDebug, "Trying again on same curl handle, "
              "same url: %d, error code %d", same_url_retry, info->error_code);
     // Reset internal state and destination
-    if ((info->destination == kDestinationMem) && info->destination_mem.data) {
-      free(info->destination_mem.data);
-      info->destination_mem.data = NULL;
-      info->destination_mem.size = 0;
-      info->destination_mem.pos = 0;
+    if (info->destination != kDestinationNone &&
+        info->destination_sink->Reset() != 0) {
+      // TODO(heretherebedragons) in theory this is unnecessary to set
+      // the error code because in the goto the flush is checked again
+      // and error_code set
```

Comment:
Under `verify_and_finalize_stop`, the error code is only set if flushing fails. So I think we should remove the comment.